src/components/DatePicker/DatePicker.js

```diff
@@ -14,6 +14,34 @@ import { IconButton } from '../IconButton'
 import languageService from '../../services/language'
 import './DatePicker.module.scss'
 
+const DatePickerTextField = props => (
+  <TextField
+    trailingIcon={
+      <IconButton
+        className={classNames('datepicker-icon', {
+          disabled: props.disabled,
+          open: props.isOpen,
+        })}
+        onClick={() => !props.disabled && props.setIsOpen(prev => !prev)}
+        size='medium'
+      >
+        <Calendar />
+      </IconButton>
+    }
+    type='date-time-picker'
+    label={props.label}
+    defaultValue={props.value}
+    onChange={props.onChange}
+    ref={props.textFieldRef}
+    format={props.format}
+    value={props.value}
+    disabled={props.disabled}
+    error={props.error}
+    message={(props.error && props.errorMessage) || props.helperText}
+    onBlur={props.onBlur}
+  />
+)
```

Comment:
separate to a different file

Reply:
```suggestion
const DatePickerTextField = ({ props, inputProps }) => (
  <TextField
    trailingIcon={
      <IconButton
        className={classNames('datepicker-icon', {
          disabled: props.disabled,
          open: props.isOpen,
        })}
        onClick={() => !props.disabled && props.setIsOpen(prev => !prev)}
        size='medium'
      >
        <Calendar />
      </IconButton>
    }
    {...inputProps}
  />
)
```
I don't think we should limit the user as to what props can be passed to the input element in inputBase. Tomorrow, if we want to pass a 'onfocusout' event, we'd have to add it here again. This way, we allow the user to do whatever without changing the code each time. What do you think?